lib/Stan_math_code_gen.ml

```diff
@@ -1,4 +1,20 @@
-(** Generate C++ from the MIR *)
+(** Generate C++ from the MIR.
+
+    This module makes extensive use of the Format[0] module via the Fmt[1] API.
+    As such, you'll need to understand the "%a" and "@" notation from [0], especially
+    the section headed "Formatted pretty-printing." Then, we use functions like
+    [pf] and [strf] from the Fmt library[1]. On top of that, the "@" pretty-printing
+    specifiers all actually correspond 1-to-1 with commands like [open_box] from
+    the Format library[0]. The boxing system is best described in this explainer
+    pdf [2], particularly Section 3 ("Format basics"). It's worth noting that someone
+    was able to make a decent-looking pretty-printer for a subset of Javascript[3] that
+    might serve as a good reference. Good luck!
+
+    [0] Format module doc: https://caml.inria.fr/pub/docs/manual-ocaml/libref/Format.html
+    [1] Fmt module doc: https://erratique.ch/software/fmt/doc/Fmt.html
+    [2] Format Unraveled: https://hal.archives-ouvertes.fr/hal-01503081/file/format-unraveled.pdf
+    [3] Javascript pretty-printer https://github.com/Virum/compiler/blob/28e807b842bab5dcf11460c8193dd5b16674951f/JavaScript.ml#L112
+*)
```

Comment:
Thank you for adding this! This really helps to get started with Format+Fmt. :-)